src/lib/proposalUtils.ts

```diff
@@ -717,6 +717,20 @@ export async function getProposalStatus(
   return "QUEUED";
 }
 
+export const proposalToCallArgs = (proposal: Proposal) => {
+  const dynamicProposalType: keyof ParsedProposalData =
```

Comment:
just a comment that I found our ParsedProposalData object hard to work with. What about you? Maybe we could refactor it at some point.